src/mm-form/mm-form.js

```diff
@@ -0,0 +1,490 @@
+
+*/
+(function (scope) {
+
+	scope.Form = Polymer({
+		is: 'mm-form',
+
+		behaviors: [
+			StrandTraits.Stylable,
+			StrandTraits.LightDomGettable,
+			StrandTraits.Resolvable,
+			StrandTraits.Columnable,
+			StrandTraits.Refable,
+			StrandTraits.Falsifiable
+		],
+
+		properties: {
+			view: {
+				type: Object
+			},
+			unsaved: {
+				type: Boolean,
+				value: true,
+				notify: true,
+				readOnly: true
+			},
+			showUnsavedMessage: {
+				type: Boolean,
+				value: true,
+				notify: true
+			},
+			autoValidate: {
+				type: Boolean,
+				value: true
+			},
+			// Footer related
+			footerMessages: {
+				type: Object,
+				value: function() {
+					return {
+						'error' : 'This form contains errors',
+						'success' : 'This form does not contain any errors',
+						'warning' : 'This form contains unsaved changes'
+					};
+				}
+			},
+			actions: {
+				type: Array,
+				value: function() {
+					return [
+						{
+							label: 'Cancel',
+							type: 'action',
+							callback: function(e,host) {
+								host.cancel();
+							}
+						},
+						{
+							label: 'Save',
+							type: 'primary',
+							callback: function(e,host) {
+								host.serializeForm();
+							}
+						} 
+					];
+				}
+			},
+			showFooterMessages: {
+				type: Boolean,
+				value: true,
+				notify: true
+			},
+			footerFixed: {
+				type: Boolean,
+				value: true,
+				notify: true
+			},
+			footerLeft: {
+				type: Number,
+				value: 0,
+				notify: true
+			},
+			_footerType: {
+				type: String,
+				notify: true
+			},
+			_footerMessage: {
+				type: String,
+				notify: true
+			},
+			_showFooterMessage: {
+				type: Boolean,
+				value: false,
+				notify: true
+			},
+			_displayFooterMessage: {
+				type: Boolean,
+				computed: '_displayMessage(_showFooterMessage, showFooterMessages)'
+			},
+			// config/initial data & settings:
+			config: {
+				type: Object,
+				observer: '_configChanged',
+				value: function() { return {}; }
+			},
+			data: {
+				type: Object,
+				observer: '_dataChanged',
+				value: function() { return {}; }
+			}
+		},
+
+		_initialData: {},
+		_invalidFields: null,
+		_validFields: null,
+
+		listeners: {
+			'changed' : '_handleChanged'
+		},
+
+		attached: function() {
+			if (this._isEmpty(this.config)) this.debounce('initConfig', this._initConfig);
+			if (this._isEmpty(this.data)) this.debounce('initData', this._initData);
+			
+			// Temp warning message
+			console.warn('This component contains experimental features. The configuration and API are subject to change. Please use at your own risk.');
+		},
+
+		_dataChanged: function(newVal, oldVal) {
+			if(!this._isEmpty(newVal)) {
+				this.debounce('initData', this._initData);
+			}
+		},
+
+		_configChanged: function(newVal, oldVal) {
+			if (!this._isEmpty(newVal)) {
+				this.debounce('initConfig', this._initConfig);
+			}
+		},
+
+		_isEmpty: function(obj) {
+			return !(Object.getOwnPropertyNames(obj).length > 0);
```

Comment:
Maybe not a use-case we're concerned with, but have you considered that fields in the prototype chain are enumerated in `for (x in y) {}` loops (and bindable in polymer)?